code/modules/flufftext/Hallucination.dm

```diff
@@ -176,30 +176,296 @@ Gunshots/explosions/opening doors/less rare audio (done)
+		dist_since_sleep++
+		if(dist_since_sleep >= speed)
+			dist_since_sleep = 0
+			sleep(1)
+
+
+	src.throwing = 0
+	src.throw_impact(get_turf(src))
+	
+	return 1
+
+/obj/effect/hallucination/simple/xeno/throw_impact(A)
+	update_icon("alienh_pounce")
+	if(A == target)
+		target.Weaken(5)
+		target << "<span class ='userdanger'>[name] pounces on you!</span>" //Todo : make this flailing for observers ?
+
+/obj/effect/hallucination/xeno_attack
+	//Xeno crawls from nearby vent,jumps at you, and goes back in
+	var/obj/machinery/atmospherics/unary/vent_pump/pump = null
+	var/obj/effect/hallucination/simple/xeno/xeno = null
+
+/obj/effect/hallucination/xeno_attack/New(loc,var/mob/living/carbon/T)
+	target = T
+	for(var/obj/machinery/atmospherics/unary/vent_pump/U in orange(7,target))
+		if(!U.welded)
+			pump = U
+			break
+	xeno = new(pump.loc,target)
+	sleep(10)
+	xeno.update_icon("alienh_leap",'icons/mob/alienleap.dmi',-32,-32)
+	xeno.throw_at(target,7,1)
+	sleep(10)
+	xeno.update_icon("alienh_leap",'icons/mob/alienleap.dmi',-32,-32)
+	xeno.throw_at(pump,7,1)
+	sleep(10)
+	var/xeno_name = xeno.name
+	target << "<span class='notice'>[xeno_name] begins climbing into the ventilation system...</span>"
+	sleep(10)
+	qdel(xeno)
+	target << "<span class='notice'>[xeno_name] scrambles into the ventilation ducts!</span>"
+	qdel(src)
+
+/obj/effect/hallucination/singularity_scare
+	//Singularity moving towards you.
+	//todo Hide where it moved with fake space images
+	var/obj/effect/hallucination/simple/singularity/s = null
+
+/obj/effect/hallucination/singularity_scare/New(loc,var/mob/living/carbon/T)
+	target = T
+	var/turf/start = T.loc
+	var/screen_border = pick(SOUTH,EAST,WEST,NORTH)
+	for(var/i = 0,i<11,i++)
+		start = get_step(start,screen_border)
+	s = new(start,target)
+	for(var/i = 0,i<11,i++)
+		sleep(5)
+		s.loc = get_step(get_turf(s),get_dir(s,target))
+		s.Show()
+		s.Eat()
+	qdel(s)
+
+/obj/effect/hallucination/simple/singularity
+	image_icon = 'icons/effects/224x224.dmi'
+	image_state = "singularity_s7"
+	image_layer = 6
+	px = -96
+	py = -96
+
+/obj/effect/hallucination/simple/singularity/proc/Eat(atom/OldLoc, Dir)
+	var/target_dist = get_dist(src,target)
+	if(target_dist<=3) //"Eaten"
+		target.sleeping = 20
+		target.hal_crit = 1
+		target.hal_screwyhud = 1
+		spawn(rand(50,100))
+			target.sleeping = 0
+			target.hal_crit = 0
+			target.hal_screwyhud = 0
+
+/obj/effect/hallucination/battle
+
+/obj/effect/hallucination/battle/New(loc,var/mob/living/carbon/T)
+	target = T
+	var/hits = rand(3,5)
+	switch(rand(1,3))
+		if(1) //Laser fight
+			for(var/i=0,i<hits,i++)
+				target << sound('sound/weapons/Laser.ogg',0,1,0,25)
+				sleep(rand(1,5))
+			target << sound(get_sfx("bodyfall"),0,1,0,25)
+		if(2) //Esword fight
+			target << sound('sound/weapons/saberon.ogg',,0,1,0,15)
+			for(var/i=0,i<hits,i++)
+				target << sound('sound/weapons/blade1.ogg',,0,1,0,25)
+				sleep(rand(1,5))
+			target << sound(get_sfx("bodyfall"),0,1,0,25)

```

Comment:
keep doing hit sounds after the body falls a few times, a proper traitor finishes the job instead of relying on crit